src/Gui/DlgSettingsWorkbenchesImp.cpp

```diff
@@ -0,0 +1,475 @@
+            if (!checked) //disabling wb disable auto-load.
+                autoloadCheckbox->setChecked(false);
+            break;
+        }
+    }
+
+    // Reset the start combo items.
+    setStartWorkbenchComboItems();
+}
+
+/**
+Build the list of unloaded workbenches.
+*/
+void DlgSettingsWorkbenchesImp::buildWorkbenchList()
+{
+    QStringList workbenches = Application::Instance->workbenches();
+    QStringList enabledWbs = getEnabledWorkbenches();
+    QStringList disabledWbs = getDisabledWorkbenches();
+
+    ui->wbList->setDragDropMode(QAbstractItemView::InternalMove);
+    ui->wbList->setSelectionMode(QAbstractItemView::SingleSelection);
+    ui->wbList->viewport()->setAcceptDrops(true);
+    ui->wbList->setDropIndicatorShown(true);
+    ui->wbList->setDragEnabled(true);
+    ui->wbList->setDefaultDropAction(Qt::MoveAction);
+
+    //First we add the enabled wbs in their saved order.
+    for (const auto& wbName : enabledWbs) {
+        if (workbenches.contains(wbName)) {
+            addWorkbench(wbName, true);
+        }
+        else {
+            Base::Console().Warning("Ignoring unknown %s workbench found in user preferences.", wbName.toStdString().c_str());
+        }
+    }
+    //Second we add workbench in alphabetical order that are either Disabled, or !enabled && !disabled, ie newly added wb.
+    for (const auto& wbName : workbenches) {
+        if (disabledWbs.contains(wbName)) {
+            addWorkbench(wbName, false);
+        }
+        else if (!enabledWbs.contains(wbName)) {
+            Base::Console().Warning("Adding unknown %s workbench.", wbName.toStdString().c_str());
+            addWorkbench(wbName, false);
+        }
+    }
+}
+
+void DlgSettingsWorkbenchesImp::addWorkbench(const QString& wbName, bool enabled)
+{
+    if (wbName.toStdString() == "NoneWorkbench")
+        return; // Do not list the default empty Workbench
+
+    QWidget* widget = createWorkbenchWidget(wbName, enabled);
+    auto wItem = new QListWidgetItem();
+    wItem->setSizeHint(widget->sizeHint());
+    ui->wbList->addItem(wItem);
+    ui->wbList->setItemWidget(wItem, widget);
+}
+
+QWidget* DlgSettingsWorkbenchesImp::createWorkbenchWidget(const QString& wbName, bool enabled)
+{
+    auto wbTooltip = Application::Instance->workbenchToolTip(wbName);
+    auto wbDisplayName = Application::Instance->workbenchMenuText(wbName);
+
+    // 1: Enable checkbox
+    auto enableCheckBox = new QCheckBox(this);
+    enableCheckBox->setToolTip(tr("If unchecked, %1 will not appear in the available workbenches.").arg(wbDisplayName));
+    enableCheckBox->setChecked(enabled);
+    enableCheckBox->setObjectName(enableCheckboxStr);
+    if (wbName.toStdString() == _startupModule) {
+        enableCheckBox->setChecked(true);
+        enableCheckBox->setEnabled(false);
+        enableCheckBox->setToolTip(tr("This is the current startup module, and must be enabled. See Preferences/General/Autoload to change."));
+    }
```

Comment:
Reading this and rest of the code, I have a doubt if the list correctly updates if "load on startup" WB is changed (didn't test).

Reply:
The list updates correctly when 'start' wb is changed.
Note, the 'start' combobox shows only the wb that are activated.